### ISSUE TITLE
Skins not being cached anymore

### ISSUE THREAD

**Issue by MrStickmanPro1** [CitizensDev/Citizens2#909]
After updating to Citizens 2.0.20 and spigot 1.10.2, cached skins are broken.

Basically all my NPCs are relying on cached skins. Nearly all skins have been set on my own account so I don't know if that is a problem. But even for the ones that were on another account, it is still showing the most up to date skin now.

Please let us know if this is fixable or if we have to find another way to make NPCs look like they are supposed to be looking again. I'd also be happy to test anything out.

Thanks in advance.

**Comment by MrStickmanPro1** [comment]
I just found this plugin: https://www.spigotmc.org/resources/changeskin.21469/

Seems like this one is able to do what we need.

**Comment by fullwall** [comment]
/npc skin -p should do it for you. What version did you update from?

**Comment by MrStickmanPro1** [comment]
Unfortunately, caching a skin with this exact command is what's not working for me. The skin always updates to the latest skin of the player.

I'll look up what version I upgraded from later on today,

It might also be worth mentioning that I have an offline mode server that is running behind an online mode spigot server.

**Comment by fullwall** [comment]
Can you please pastebin your saves.yml?

**Comment by MrStickmanPro1** [comment]
Sure thing. Here it is: http://pastebin.com/72JukCgU

When base64-decoding the skin data, you will notice that most of these skins were set on my own account (Mike_x1). That is because I applied most of them to my own account, made Citizens cache them and then changed my skin again.

**Comment by MrStickmanPro1** [comment]
Any updates on this so far? I just tried setting my bungeecord server to online mode. However, it still didn't work (still seeing my current skin for all npcs using a cached skin on my account).

**Comment by MrStickmanPro1** [comment]
Another update on this: After comparing the differences to some test results on a completely new server with just citizens on it (where it worked), I managed to make it work on my actual server as well.

This is what I did in the config (on each NPC with a custom skin):
- add `player-skin-use-latest: false` to the metadata config section
- make the value of `cached-skin-uuid-name` lowercase

---

Since I just noticed that I didn't answer this yet: I updated from `2.0.15-SNAPSHOT (build 1228)`.

Anyway... Now that this works, I have found another issue (see #933).
